src/main/java/com/maddyhome/idea/vim/extension/argtextobj/VimArgTextObjExtension.java

```diff
@@ -298,33 +298,52 @@ private static class ArgBoundsFinder {
      * @param position starting position.
      */
     boolean findBoundsAt(int position) throws IllegalStateException {
-      if (text.isEmpty()) {
+      if (text.length() == 0) {
         error = "empty document";
         return false;
       }
+
+      // First try original position
       leftBound = Math.min(position, leftBound);
       rightBound = Math.max(position, rightBound);
+
+      // Get out of quoted text and handle any current position
       getOutOfQuotedText();
+      char ch = getCharAt(rightBound);
       if (rightBound == leftBound) {
-        if (brackets.isCloseBracket(getCharAt(rightBound))) {
+        if (ch == 0) {
+          error = "invalid position";
+          return false;
+        }
+        if (brackets.isCloseBracket(ch)) {
           --leftBound;
+        } else if (brackets.isOpenBracket(ch)) {
+          ++rightBound;
```

Comment:
Th else branch will do the same, do we need this?

Reply:
Good point. Fixed.